base/intset.jl

```diff
@@ -68,12 +68,11 @@ function push!(s::IntSet, n::Integer)
     return s
 end
 
-function union!(s::IntSet, ns)
-    for n in ns
-        push!(s, n)
-    end
-    return s
-end
+union!(s::AbstractSet, xs) = _union!(s, xs)
+
+_union!(s::AbstractSet, xs) = rawunion!(s, xs)
```

Comment:
`union!` for `Set` types needs a size hint before calling `rawunion!`, whereas in `IntSet` one gets a size hint for every `push!` (as the underlying container size depends on largest value).